syntax = "proto3";
package solutions;

import "tptp.proto";
import "google/protobuf/duration.proto";
import "google/protobuf/timestamp.proto";

message Report {
  google.protobuf.Timestamp date = 1;
  string commit = 2;
  repeated string labels = 3;
  repeated Case cases = 4;
}

message Case {
  string name = 1;
  google.protobuf.Duration duration = 2;
  tptp.File fof_problem = 3;
  tptp.File cnf_problem = 4;
  ProverOutput output = 5;
}

message ProverOutput {
  int64 cost = 1;
  int64 leaf_count = 2;
  int64 extension_count = 3;
  tptp.File proof = 4;
}

message Stats {
  int64 refl = 1;
  int64 symm = 2;
  int64 trans = 3;

  message PredMono { string name = 1; int64 count = 2; };
  message FunMono { string name = 1; int64 count = 2; };
  repeated PredMono pred_mono = 4;
  repeated FunMono fun_mono = 5;

  message OrClause { tptp.Input cla = 1; int64 count = 2; };
  repeated OrClause or_clauses = 6;
}

// deprecated
message CNF {
  tptp.File problem = 1;
  tptp.File proof = 2;
}

